src/ol/array.js

```diff
@@ -126,15 +126,13 @@ export function reverseSubArray(arr, begin, end) {
 
 /**
  * @param {Array.<VALUE>} arr The array to modify.
- * @param {Array.<VALUE>|VALUE} data The elements or arrays of elements
- *     to add to arr.
+ * @param {Array.<VALUE>|VALUE} data The elements or arrays of elements to add to arr.
  * @template VALUE
  */
 export function extend(arr, data) {
-  let i;
-  const extension = Array.isArray(data) ? data : [data];
+  const extension = /** @ type {Array.<VALUE> */ (Array.isArray(data) ? data : [data]);
```

Comment:
Is this needed? And if so, shouldn't it be `@type` instead of `@ type`?

Reply:
Yes, that's an error. Instead of a type cast, the warning can be fixed by changing the `data` type to `{!Array.<VALUE>|VALUE}`

The compiler warning is:
```
./build/src-closure/ol/array.js:136: WARNING - The right side in the assignment is not a subtype of the left side.
Expected : VALUE
Found    : (VALUE|null)
More details:
The found type is a union that includes an unexpected type: null
    arr[arr.length] = extension[i];
```